### PR TITLE
Allow custom (non-propagating) stateful set metadata

### PR DESCRIPTION
**Pull request by ramondeklein**
This PR fixes https://github.com/minio/operator/issues/2318. This PR will remove all automatic propagation of metadata from the `Tenant` resource to the child resources (statefulset, pod, service, pvc).

- Each pool results in a statefulset. Labels/annotations that should be used for all pools, should be set using `.spec.poolsMetadata.Annotations` / `.spec.poolsMetadata.Labels`. Per-pool labels/annotations can be set using `.spec.pools[].Annotations` or `.spec.pools[].Labels`.
- The (non-headless) services can be annotated with the fields in `.spec.serviceMetadata`. There are custom options for MinIO, console and KES.
- PVCs are specified using a generic `PersistentVolumeClaim` object.

## Breaking change
Note that this may be a breaking change for some deployments that depend on propagating annotations/labels from the tenant to downstream resources.

### CODE REVIEW

File: pkg/client/applyconfiguration/minio.min.io/v2/statefulsetmetadata.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// Code generated by applyconfiguration-gen. DO NOT EDIT.
+
+package v2
+
+// StatefulSetMetadataApplyConfiguration represents an declarative configuration of the StatefulSetMetadata type for use
```
Comment (shtripat) [inline pkg/client/applyconfiguration/minio.min.io/v2/statefulsetmetadata.go:21]:
```suggestion
// StatefulSetMetadataApplyConfiguration represents a declarative configuration of the StatefulSetMetadata type for use
```

File: pkg/client/applyconfiguration/minio.min.io/v2/statefulsetmetadata.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+// with apply.
+type StatefulSetMetadataApplyConfiguration struct {
+	Labels      map[string]string `json:"labels,omitempty"`
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+// StatefulSetMetadataApplyConfiguration constructs an declarative configuration of the StatefulSetMetadata type for use with
```
Comment (shtripat) [inline pkg/client/applyconfiguration/minio.min.io/v2/statefulsetmetadata.go:28]:
```suggestion
// StatefulSetMetadataApplyConfiguration constructs a declarative configuration of the StatefulSetMetadata type for use with
```

**Review by shtripat — DISMISSED** [review]
Mostly looks good. Couple of minor ones.

File: pkg/client/applyconfiguration/minio.min.io/v2/poolsmetadata.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// Code generated by applyconfiguration-gen. DO NOT EDIT.
+
+package v2
+
+// PoolsMetadataApplyConfiguration represents an declarative configuration of the PoolsMetadata type for use
```
Comment (shtripat) [inline pkg/client/applyconfiguration/minio.min.io/v2/poolsmetadata.go:21]:
```suggestion
// PoolsMetadataApplyConfiguration represents a declarative configuration of the PoolsMetadata type for use
```

Comment (ramondeklein) [reply pkg/client/applyconfiguration/minio.min.io/v2/poolsmetadata.go:21]: This typo is automatically generated by `k8s.io/code-generator`. I can't do anything about that.

File: pkg/client/applyconfiguration/minio.min.io/v2/statefulsetmetadata.go (hunk 1 above)
Comment (ramondeklein) [reply pkg/client/applyconfiguration/minio.min.io/v2/statefulsetmetadata.go]: Automatically generated, so can't do anything about it.

**Comment by ramondeklein** [comment]
@shtripat I'll update the code generator in another PR to fix the spelling issue. It annoyed me too 😄

**Review by shtripat — DISMISSED** [review]
lgtm

**Comment by ramondeklein** [comment]
@cesnietor Done...

**Review by cesnietor — APPROVED** [review]

**Review by pjuarezd — APPROVED** [review]

**Comment by acelinkio** [comment]
Any chance a new release could be cut to catch this change?  Thanks!

**Pull request merged**